packages/gatsby-theme-newrelic/src/components/SwiftSearch.js

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import SiteSearchAPIConnector from '@elastic/search-ui-site-search-connector';
+import {
+  SearchProvider,
+  WithSearch,
+  SearchBox,
+  Results,
+  Paging,
+  PagingInfo,
+} from '@elastic/react-search-ui';
+import ResultView from './ResultView';
+import PagingInfoView from './PagingInfoView';
+import SearchInput from './SearchInput';
+
+import styled from '@emotion/styled';
+import { css } from '@emotion/core';
+import PropTypes from 'prop-types';
+
+// TODO: move styles to emotion/wrapper
+import Icon from './Icon';
+import styles from '../styles/SwiftSearchStyles';
+
+const connector = new SiteSearchAPIConnector({
+  documentType: 'page',
+  engineKey: 'Ad9HfGjDw4GRkcmJjUut',
+});
+
+const configOptions = {
+  apiConnector: connector,
+  searchQuery: {
+    result_fields: {
+      title: {
+        snippet: {
+          size: 100,
+          fallback: true,
+        },
+      },
+      body: {
+        snippet: {
+          size: 400,
+          fallback: true,
+        },
+      },
+      url: {
+        raw: {},
+      },
+    },
+  },
+  initialState: {
+    resultsPerPage: 10,
+  },
+};
+
+const SwiftSearch = ({ className }) => {
+  return (
+    <div css={styles} className={className}>
+      <SearchProvider config={configOptions}>
+        <WithSearch
+          mapContextToProps={({ isLoading, results, searchTerm }) => ({
```

Comment:
Out of curiosity, is this not the default? Without looking at the documentation (maybe I should 😄 ), is this not the default? My guess is this gets mapped into the callback on [line 65](https://github.com/newrelic/gatsby-theme-newrelic/pull/59/files#diff-f3e8ad67873c15cac146aaeff71f04cfR65)?

Could you maybe talk through this and way this is necessary to map props back into an object of the same shape.

Reply:
i'm not entirely sure why they're doing it this way but we were going off the example in their docs [here](https://github.com/elastic/search-ui/blob/master/ADVANCED.md#mapcontexttoprops)

Reply:
Oh ok! It looks like maybe a performance optimization if I'm reading that right. Thanks for pointing me there!